src/test/architecture/test_architecture.py

```diff
@@ -312,3 +312,146 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
+  executor_name: executor_1
+  callback_group_names:
+  - /callback_group_0
+  - /callback_group_1"""
+
+        # define test case: node, callback, topic
+        architecture_text += """
+nodes:
+- node_name: /node_0
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_1
+    callback_names:
+    - /callback_0
+    - /callback_1
+  callbacks:
+  - callback_name: /callback_0
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_1
+    callback_type: subscription_callback
+    topic_name: /topic_1
+    symbol: sub
+  publishes:
+  - topic_name: /topic_0
+    callback_names:
+    - /callback_0
+  subscribes:
+  - topic_name: /topic_1
+    callback_name: /callback_1"""
+
+        # define test case: node, callback, topic
+        architecture_text += """
+- node_name: /node_1
+  callback_groups:
+  - callback_group_type: mutually_exclusive
+    callback_group_name: /callback_group_0
+    callback_names:
+    - /callback_2
+    - /callback_3
+  callbacks:
+  - callback_name: /callback_2
+    callback_type: timer_callback
+    period_ns: 200000000
+    symbol: timer
+  - callback_name: /callback_3
+    callback_type: subscription_callback
+    topic_name: /topic_0
+    symbol: sub
+  publishes:
+  - topic_name: /topic_1
+    callback_names:
+    - /callback_2
+  subscribes:
+  - topic_name: /topic_0
+    callback_name: /callback_3
+        """
+
+        mocker.patch('builtins.open', mocker.mock_open(read_data=architecture_text))
+        arch = Architecture('yaml', 'architecture.yaml')
+
+        # test rename_node()
+        node_names = arch.node_names
+        expect_node_names = ['/node_0', '/node_1']
+        assert set(node_names) == set(expect_node_names)
+
+        arch.rename_node('/node_1', '/changed_node')
+
+        node_names = arch.node_names
+        expect_node_names = ['/node_0', '/changed_node']
+        assert set(node_names) == set(expect_node_names)
+
+        # test rename_executor()
+        executor_names = arch.executor_names
+        expect_executor_names = ['executor_0', 'executor_1']
+        assert set(executor_names) == set(expect_executor_names)
+
+        arch.rename_executor('executor_1', 'changed_executor')
+
+        executor_names = arch.executor_names
+        expect_executor_names = ['executor_0', 'changed_executor']
+        assert set(executor_names) == set(expect_executor_names)
+
+        # test rename_topic()
+        topic_names = arch.topic_names
+        expect_topic_names = ['/topic_0', '/topic_1']
+        # assert set(topic_names) == set(expect_topic_names)
+
+        arch.rename_topic('/topic_1', '/changed_topic')
+
+        topic_names = arch.topic_names
+        expect_topic_names = ['/topic_0', '/changed_topic']
+        assert set(topic_names) == set(expect_topic_names)
+
+        # test rename_callback()
+        callback_names = [c.callback_name for c in arch.callbacks]
```

Comment:
Confirmation is required for all related instances.
The Struct class should be changed from Tuple to List.

```
        assert arch.callbacks[0].publish_topic_names == ('/changed_topic',)
```

If the test becomes too large, it may be a good idea to divide the test cases by class.

Reply:
Fixed it in [52c0a76](https://github.com/tier4/CARET_analyze/pull/156/commits/52c0a76f97cb01724b0587f1b0075a5296502dd1), [437218c](https://github.com/tier4/CARET_analyze/pull/156/commits/437218c14b36fb5a1cfd0df3b453751bfc39c289), [4709c6d](https://github.com/tier4/CARET_analyze/pull/156/commits/4709c6d57b92cebb8adfb67ef644b9e2f630f775), and [2fec610](https://github.com/tier4/CARET_analyze/pull/156/commits/2fec6102a4603e0bf38a7e83a9dbd57cf14915b0)